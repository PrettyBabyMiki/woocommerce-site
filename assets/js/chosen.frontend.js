jQuery(function(){

	// Frontend Chosen selects
	jQuery("select.country_select, select.state_select").chosen();
	
	jQuery('body').bind('country_to_state_changed', function(){
		jQuery("select.state_select").chosen().trigger("liszt:updated");
	});
	
});