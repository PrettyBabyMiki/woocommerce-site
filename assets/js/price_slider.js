jQuery(document).ready(function($) {
	
	// Get markup ready for slider
	$('input#min_price, input#max_price').hide();
	$('.price_slider, .price_label').show();
	
	// Price slider uses jquery ui
	var min_price = $('.price_slider_amount #min_price').attr('data-min');
	var max_price = $('.price_slider_amount #max_price').attr('data-max');
	
	current_min_price = parseInt(min_price);
	current_max_price = parseInt(max_price);
	
	if (woocommerce_price_slider_params.min_price) current_min_price = parseInt(woocommerce_price_slider_params.min_price);
	if (woocommerce_price_slider_params.max_price) current_max_price = parseInt(woocommerce_price_slider_params.max_price);
	
	function update_price_span( min, max ) {
	
		if (woocommerce_price_slider_params.currency_pos == "left") {
		
			$( ".price_slider_amount span.from" ).html( woocommerce_price_slider_params.currency_symbol + min );
			$( ".price_slider_amount span.to" ).html( woocommerce_price_slider_params.currency_symbol + max );
			
		} else if (woocommerce_price_slider_params.currency_pos == "left_space") {
			
			$( ".price_slider_amount span.from" ).html( woocommerce_price_slider_params.currency_symbol + " " + min );
			$( ".price_slider_amount span.to" ).html( woocommerce_price_slider_params.currency_symbol + " " + max );
			
		} else if (woocommerce_price_slider_params.currency_pos == "right") {
			
			$( ".price_slider_amount span.from" ).html( min + woocommerce_price_slider_params.currency_symbol );
			$( ".price_slider_amount span.to" ).html( max + woocommerce_price_slider_params.currency_symbol );
			
		} else if (woocommerce_price_slider_params.currency_pos == "right_space") {
			
			$( ".price_slider_amount span.from" ).html( min + " " + woocommerce_price_slider_params.currency_symbol );
			$( ".price_slider_amount span.to" ).html( max + " " + woocommerce_price_slider_params.currency_symbol );
			
		}
			
	}
	
	$('.price_slider').slider({
		range: true,
		animate: true,
		min: min_price,
		max: max_price,
		values: [current_min_price,current_max_price],
		create : function( event, ui ) {
			
			update_price_span(current_min_price, current_max_price);
			
			$( ".price_slider_amount #min_price" ).val(current_min_price);
			$( ".price_slider_amount #max_price" ).val(current_max_price);
			
			$('body').trigger('price_slider_create');
		},
		slide: function( event, ui ) {
			
			update_price_span(ui.values[ 0 ], ui.values[ 1 ]);

			$( "input#min_price" ).val(ui.values[ 0 ]);
			$( "input#max_price" ).val(ui.values[ 1 ]);
			
			$('body').trigger('price_slider_slide');
		}
	});
	
});