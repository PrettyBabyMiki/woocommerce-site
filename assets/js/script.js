jQuery.fn.animateHighlight = function(highlightColor, duration) {
    var highlightBg = highlightColor || "#FFFF9C";
    var animateMs = duration || 1500;
    var originalBg = this.css("backgroundColor");
    this.stop().css("background-color", highlightBg).animate({backgroundColor: originalBg}, animateMs);
};

jQuery(function(){
	
	// Lightbox
	jQuery('a.zoom').fancybox({
		'transitionIn'	:	'elastic',
		'transitionOut'	:	'elastic',
		'speedIn'		:	600, 
		'speedOut'		:	200, 
		'overlayShow'	:	true
	});
	
	// Star ratings
	jQuery('#rating').hide().before('<p class="stars"><span><a class="star-1" href="#">1</a><a class="star-2" href="#">2</a><a class="star-3" href="#">3</a><a class="star-4" href="#">4</a><a class="star-5" href="#">5</a></span></p>');
	
	jQuery('p.stars a').click(function(){
		jQuery('#rating').val(jQuery(this).text());
		jQuery('p.stars a').removeClass('active');
		jQuery(this).addClass('active');
		return false;
	});

	// Price slider
	var min_price = jQuery('.price_slider_amount #min_price').val();
	var max_price = jQuery('.price_slider_amount #max_price').val();
	
	if (params.min_price) {
		current_min_price = params.min_price;
	} else {
		current_min_price = min_price;
	}
	
	if (params.max_price) {
		current_max_price = params.max_price;
	} else {
		current_max_price = max_price;
	}
	
	jQuery('.price_slider').slider({
		range: true,
		min: min_price,
		max: max_price,
		values: [ current_min_price, current_max_price ],
		create : function( event, ui ) {
			jQuery( ".price_slider_amount span" ).html( params.currency_symbol + current_min_price + " - " + params.currency_symbol + current_max_price );
			jQuery( ".price_slider_amount #min_price" ).val(current_min_price);
			jQuery( ".price_slider_amount #max_price" ).val(current_max_price);
		},
		slide: function( event, ui ) {
			jQuery( ".price_slider_amount span" ).html( params.currency_symbol + ui.values[ 0 ] + " - " + params.currency_symbol + ui.values[ 1 ] );
			jQuery( "input#min_price" ).val(ui.values[ 0 ]);
			jQuery( "input#max_price" ).val(ui.values[ 1 ]);
		}
	});
			
	// Quantity buttons
	jQuery("div.quantity, td.quantity").append('<input type="button" value="+" id="add1" class="plus" />').prepend('<input type="button" value="-" id="minus1" class="minus" />');
	jQuery(".plus").click(function()
	{
	    var currentVal = parseInt(jQuery(this).prev(".qty").val());
	   
	    if (!currentVal || currentVal=="" || currentVal == "NaN") currentVal = 0;
	    
	    jQuery(this).prev(".qty").val(currentVal + 1); 
	});
	
	jQuery(".minus").click(function()
	{
	    var currentVal = parseInt(jQuery(this).next(".qty").val());
	    if (currentVal == "NaN") currentVal = 0;
	    if (currentVal > 0)
	    {
	        jQuery(this).next(".qty").val(currentVal - 1);
	    }
	});
	
	/* states */
	var states_json = params.countries.replace(/&quot;/g, '"');
	var states = jQuery.parseJSON( states_json );			
	
	jQuery('select.country_to_state').change(function(){
		
		var country = jQuery(this).val();
		var state_box = jQuery('#' + jQuery(this).attr('rel'));
		
		var input_name = jQuery(state_box).attr('name');
		var input_id = jQuery(state_box).attr('id');

		if (states[country]) {
			var options = '';
			var state = states[country];
			for(var index in state) {
				options = options + '<option value="' + index + '">' + state[index] + '</option>';
			}
			if (jQuery(state_box).is('input')) {
				// Change for select
				jQuery(state_box).replaceWith('<select name="' + input_name + '" id="' + input_id + '"><option value="">' + params.select_state_text + '</option></select>');
				state_box = jQuery('#' + jQuery(this).attr('rel'));
			}
			jQuery(state_box).append(options);
		} else {
			if (jQuery(state_box).is('select')) {
				jQuery(state_box).replaceWith('<input type="text" placeholder="' + params.state_text + '" name="' + input_name + '" id="' + input_id + '" />');
				state_box = jQuery('#' + jQuery(this).attr('rel'));
			}
		}
		
	}).change();
	
	/* Tabs */
	jQuery('#tabs .panel:not(#tabs .panel)').hide();
	jQuery('#tabs li a').click(function(){
		var href = jQuery(this).attr('href');
		jQuery('#tabs li').removeClass('active');
		jQuery('div.panel').hide();
		jQuery('div' + href).show();
		jQuery(this).parent().addClass('active');
		jQuery.cookie('current_tab', href);
		return false;
	});
	if (jQuery('#tabs li.active').size()==0) {
		jQuery('#tabs li:first a').click();
	} else {
		jQuery('#tabs li.active a').click();
	}
	
	/* Shipping calculator */
	
	jQuery('.shipping-calculator-form').hide();
	
	jQuery('.shipping-calculator-button').click(function() {
	  jQuery('.shipping-calculator-form').slideToggle('slow', function() {
	    // Animation complete.
	  });
	}); 
	
	// Stop anchors moving the viewport

	jQuery(".shipping-calculator-button").click(function() { return false; });
	
	// Variations
	
	function check_variations() {
		
		var not_set = false;
		
		jQuery('.variations select').each(function(){
			if (jQuery(this).val()=="") not_set = true;
		});
		
		jQuery('.variations_button, .single_variation').slideUp();
		
		if (!not_set) {
			
			jQuery('.variations').block({ message: null, overlayCSS: { background: '#fff url(' + params.plugin_url + '/assets/images/ajax-loader.gif) no-repeat center', opacity: 0.6 } });
			
			var data = {
				action: 		'jigoshop_get_variation',
				variation_data: jQuery('form.variations_form').serialize(),
				security: 		params.get_variation_nonce
			};

			jQuery.post( params.ajax_url, data, function(response) {
				
				var img = jQuery('div.images img:eq(0)');
				var link = jQuery('div.images a.zoom');
				var o_src = jQuery(img).attr('original-src');
				var o_link = jQuery(link).attr('original-href');

				if (response.length > 1) {
				
					variation_response = jQuery.parseJSON( response );
					
					var variation_image = variation_response.image_src;
					var variation_link = variation_response.image_link;

					jQuery('.single_variation').html( variation_response.price_html + variation_response.availability_html );
					
					if (!o_src) {
						jQuery(img).attr('original-src', jQuery(img).attr('src'));
					}
					
					if (!o_link) {
						jQuery(link).attr('original-href', jQuery(link).attr('href'));
					}
					
					if (variation_image.length > 1) {	
						jQuery(img).attr('src', variation_image);
						jQuery(link).attr('href', variation_link);
					} else {
						jQuery(img).attr('src', o_src);
						jQuery(link).attr('href', o_link);
					}

					jQuery('.variations_button, .single_variation').slideDown();
				} else {
					if (o_src) {
						jQuery(img).attr('src', o_src);
						jQuery(link).attr('href', o_link);
					}
					jQuery('.single_variation').slideDown();
					jQuery('.single_variation').html( '<p>' + params.variation_not_available_text + '</p>' );
				}
								
				jQuery('.variations').unblock();
			});
		
		} else {
			jQuery('.variations_button').hide();
		}
		
	}
	
	jQuery('.variations select').change(function(){
		check_variations();
	});
	
});

if (params.is_checkout==1) {

	var updateTimer;
	
	function update_checkout() {
	
		var method = jQuery('#shipping_method').val();
		
		var country 	= jQuery('#billing-country').val();
		var state 		= jQuery('#billing-state').val();
		var postcode 	= jQuery('input#billing-postcode').val();
			
		if (jQuery('#shiptobilling input').is(':checked') || jQuery('#shiptobilling input').size()==0) {
			var s_country 	= jQuery('#billing-country').val();
			var s_state 	= jQuery('#billing-state').val();
			var s_postcode 	= jQuery('input#billing-postcode').val();
			
		} else {
			var s_country 	= jQuery('#shipping-country').val();
			var s_state 	= jQuery('#shipping-state').val();
			var s_postcode 	= jQuery('input#shipping-postcode').val();
		}
		
		jQuery('#order_methods, #order_review').block({ message: null, overlayCSS: { background: '#fff url(' + params.plugin_url + '/assets/images/ajax-loader.gif) no-repeat center', opacity: 0.6 } });
		jQuery.ajax({
			type: 		'POST',
			url: 		params.review_order_url,
			data: 		{ shipping_method: method, country: country, state: state, postcode: postcode, s_country: s_country, s_state: s_state, s_postcode: s_postcode },
			success: 	function( code ) {
							jQuery('#order_methods, #order_review').remove();
							jQuery('#order_review_heading').after(code);
							jQuery('#order_review input[name=payment_method]:checked').click();
						},
			dataType: 	"html"
		});
	
	}
		
	jQuery(function(){
		
		jQuery('p.password').hide();
		
		jQuery('input.show_password').change(function(){
			jQuery('p.password').slideToggle();
		});
		
		jQuery('div.shipping-address').hide();
		
		jQuery('#shiptobilling input').change(function(){
			jQuery('div.shipping-address').hide();
			if (!jQuery(this).is(':checked')) {
				jQuery('div.shipping-address').slideDown();
			}
		}).change();
		
		if (params.option_guest_checkout=='yes') {
			
			jQuery('div.create-account').hide();
			
			jQuery('input#createaccount').change(function(){
				jQuery('div.create-account').hide();
				if (jQuery(this).is(':checked')) {
					jQuery('div.create-account').slideDown();
				}
			}).change();
		
		}
		
		jQuery('.payment_methods input.input-radio').live('click', function(){
			jQuery('div.payment_box').hide();
			if (jQuery(this).is(':checked')) {
				jQuery('div.payment_box.' + jQuery(this).attr('ID')).slideDown();
			}
		});
		
		jQuery('#order_review input[name=payment_method]:checked').click();
		
		jQuery('form.login').hide();
		
		jQuery('a.showlogin').click(function(){
			jQuery('form.login').slideToggle();
		});
		
		/* Update totals */
		jQuery('#shipping_method').live('change', function(){
			clearTimeout(updateTimer);
			update_checkout();
		}).change();
		jQuery('input#billing-country, input#billing-state, #billing-postcode, input#shipping-country, input#shipping-state, #shipping-postcode').live('keydown', function(){
			clearTimeout(updateTimer);
			updateTimer = setTimeout("update_checkout()", '1000');
		});
		jQuery('select#billing-country, select#billing-state, select#shipping-country, select#shipping-state, #shiptobilling input').live('change', function(){
			clearTimeout(updateTimer);
			update_checkout();
		});
		
		/* AJAX Form Submission */
		jQuery('form.checkout').submit(function(){
			var form = this;
			jQuery(form).block({ message: null, overlayCSS: { background: '#fff url(' + params.plugin_url + '/assets/images/ajax-loader.gif) no-repeat center', opacity: 0.6 } });
			jQuery.ajax({
				type: 		'POST',
				url: 		params.checkout_url,
				data: 		jQuery(form).serialize(),
				success: 	function( code ) {
								jQuery('.jigoshop_error, .jigoshop_message').remove();
								try {
									success = jQuery.parseJSON( code );					
									window.location = decodeURI(success.redirect);
								}
								catch(err) {
								  	jQuery(form).prepend( code );
									jQuery(form).unblock(); 
									jQuery.scrollTo(jQuery(form).parent(), { easing:'swing' });
								}
							},
				dataType: 	"html"
			});
			return false;
		});
	
	});
	
}