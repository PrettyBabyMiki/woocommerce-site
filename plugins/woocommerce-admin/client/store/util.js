/** @format */

/**
 * Returns a string representation of a sorted query object.
 *
 * @param {Object} query Current state
 * @return {String}       Query Key
 */

export function getJsonString( query = {} ) {
	return JSON.stringify( query, Object.keys( query ).sort() );
}
