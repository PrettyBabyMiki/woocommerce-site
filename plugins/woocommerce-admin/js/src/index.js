/** @format */
/**
 * External dependencies
 */
import { createElement, render } from '@wordpress/element';
import { addFilter } from '@wordpress/hooks';

/**
 * Internal dependencies
 */
import Dashboard from './dashboard';

render(
	createElement( Dashboard ),
	document.getElementById( 'root' )
);

setTimeout( function() {
	function editText( string ) {
		return `Also filtered: ${ string }`;
	}
	addFilter( 'woodash.example', 'editText', editText );
}, 1500 );

function editText2( string ) {
	return `Filtered: ${ string }`;
}
addFilter( 'woodash.example2', 'editText2', editText2 );
