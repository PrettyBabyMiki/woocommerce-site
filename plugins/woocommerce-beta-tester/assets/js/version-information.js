jQuery(function( $ ) {

	/**
	 * Version information
	 */
	var wc_beta_tester_version_information = {

		/**
		 * Initialize Version Information click
		 */
		init: function() {
			$( '#wp-admin-bar-show-version-info' )
				.on( 'click', this.showModal );
		},

		/**
		 * Handler for showing/hiding version information modal
		 */
		showModal: function( event ) {
			event.preventDefault();

			$( this ).WCBackboneModal({
				template: 'wc-beta-tester-version-info',
				variable: {
					version: wc_beta_tester_version_info_params.version,
					description: wc_beta_tester_version_info_params.description,
				},
			});
		}
	};

	wc_beta_tester_version_information.init();
});
