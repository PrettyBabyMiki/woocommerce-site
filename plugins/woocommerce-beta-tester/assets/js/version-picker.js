jQuery(function( $ ) {

	/**
	 * Version information
	 */
	var wc_beta_tester_version_picker = {

		/**
		 * Initialize Version Information click
		 */
		init: function() {
			instance = this;
			instance.new_version = undefined;

			$( '#wcbt-modal-version-switch-confirm' )
				.on( 'click', this.showConfirmVersionSwitchModal );
			$( 'input[type=radio][name=wcbt_switch_to_version]' ).change( function() {
				if ( $( this ).is( ':checked' ) ) {
					instance.new_version = $( this ).val();
				}
			} ).trigger( 'change' );
		},

		/**
		 * Handler for showing/hiding version switch modal
		 */
		showConfirmVersionSwitchModal: function( event ) {
			event.preventDefault();

			if ( ! instance.new_version ) {
				alert( wc_beta_tester_version_picker_params.i18n_pick_version );
			} else {
				$( this ).WCBackboneModal({
					template: 'wcbt-version-switch-confirm',
					variable: {
						new_version: instance.new_version,
					},
				});

				$( '#wcbt-submit-version-switch' )
					.on( 'click', instance.submitSwitchVersionForm );
			}
		},

		/**
		 * Submit form to switch version of WooCommerce.
		 */
		submitSwitchVersionForm: function( event ) {
			event.preventDefault();

			$( 'form[name=wcbt-select-version]' ).get(0).submit();
		},
	};

	wc_beta_tester_version_picker.init();

});
