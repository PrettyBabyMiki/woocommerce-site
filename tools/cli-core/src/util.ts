/**
 * External dependencies
 */
import { createServer, Server } from 'net';
import { join } from 'path';
import { writeFile } from 'fs/promises';
import { exec } from 'child_process';
import { promisify } from 'util';

export const execAsync = promisify( exec );

/**
 * Format version string for regex.
 *
 * @param {string} rawVersion Raw version number.
 * @return {string} version regex.
 */
export const getVersionRegex = ( rawVersion: string ): string => {
	const version = rawVersion.replace( /\./g, '\\.' );

	if ( rawVersion.endsWith( '.0' ) ) {
		return version + '|' + version.slice( 0, -3 ) + '\\n';
	}

	return version;
};

/**
 * Get filename from patch
 *
 * @param {string} str String to extract filename from.
 * @return {string} formatted filename.
 */
export const getFilename = ( str: string ): string => {
	return str.replace( /^a(.*)\s.*/, '$1' );
};

/**
 * Get patches
 *
 * @param {string} content Patch content.
 * @param {RegExp} regex   Regex to find specific patches.
 * @return {string[]} Array of patches.
 */
export const getPatches = ( content: string, regex: RegExp ): string[] => {
	const patches = content.split( 'diff --git ' );
	const changes: string[] = [];

	for ( const p in patches ) {
		const patch = patches[ p ];
		const id = patch.match( regex );

		if ( id ) {
			changes.push( patch );
		}
	}

	return changes;
};

/**
 * Determine if the default port for wp-env is already taken. If so, see
 * https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/#2-check-the-port-number
 * for alternatives.
 *
 * @return {Promise<boolean>} if the port is being currently used.
 */
export const isWPEnvPortTaken = () => {
	return new Promise< boolean >( ( resolve, reject ) => {
		const test: Server = createServer()
			.once( 'error', ( err: { code: string } ) => {
				return err.code === 'EADDRINUSE'
					? resolve( true )
					: reject( err );
			} )
			.once( 'listening', () => {
				return test.once( 'close', () => resolve( false ) ).close();
			} )
			.listen( '8888' );
	} );
};

/**
 * Start wp-env.
 *
 * @param {string}   tmpRepoPath - path to the temporary repo to start wp-env from.
 * @param {Function} error       -  error print method.
 * @return {boolean} if starting the container succeeded.
 */
export const startWPEnv = async (
	tmpRepoPath: string,
	error: ( s: string ) => void
) => {
	try {
		// Stop wp-env if its already running.
		await execAsync( 'wp-env stop', {
			cwd: join( tmpRepoPath, 'plugins/woocommerce' ),
			encoding: 'utf-8',
		} );
	} catch ( e ) {
		// If an error is produced here, it means wp-env is not initialized and therefore not running already.
	}

	try {
		if ( await isWPEnvPortTaken() ) {
			throw new Error(
				'Unable to start wp-env. Make sure port 8888 is available or specify port number WP_ENV_PORT in .wp-env.override.json'
			);
		}

		await execAsync( 'wp-env start', {
			cwd: join( tmpRepoPath, 'plugins/woocommerce' ),
			encoding: 'utf-8',
		} );
		return true;
	} catch ( e ) {
		let message = '';
		if ( e instanceof Error ) {
			message = e.message;
			error( message );
		}
		return false;
	}
};

/**
 * Stop wp-env.
 *
 * @param {string}   tmpRepoPath - path to the temporary repo to stop wp-env from.
 * @param {Function} error       - error print method.
 * @return {boolean} if stopping the container succeeded.
 */
export const stopWPEnv = async (
	tmpRepoPath: string,
	error: ( s: string ) => void
): Promise< boolean > => {
	try {
		await execAsync( 'wp-env stop', {
			cwd: join( tmpRepoPath, 'plugins/woocommerce' ),
			encoding: 'utf-8',
		} );
		return true;
	} catch ( e ) {
		let message = '';
		if ( e instanceof Error ) {
			message = e.message;
			error( message );
		}
		return false;
	}
};

/**
 * Generate a JSON file with the data passed.
 *
 * @param  filePath - path to the file to be created.
 * @param  data     - data to be written to the file.
 * @return {Promise<void>} - promise that resolves when the file is written.
 */
export const generateJSONFile = ( filePath: string, data: unknown ) => {
	const json = JSON.stringify(
		data,
		function replacer( _, value ) {
			if ( value instanceof Map ) {
				return Array.from( value.entries() );
			}
			return value;
		},
		2
	);
	return writeFile( filePath, json );
};
