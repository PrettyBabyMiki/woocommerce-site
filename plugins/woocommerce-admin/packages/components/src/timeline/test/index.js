/* eslint-disable jest/no-mocks-import */
/**
 * External dependencies
 */
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import Timeline from '..';
import mockData from '../__mocks__/timeline-mock-data';
import { groupItemsUsing, sortByDateUsing } from '../util.js';

describe( 'Timeline', () => {
	test( 'Renders empty correctly', () => {
		const timeline = shallow( <Timeline /> );
		expect( timeline.find( '.timeline_no_events' ).length ).toBe( 1 );
		expect(
			timeline
				.find( '.timeline_no_events' )
				.first()
				.contains( 'No data to display' )
		).toBe( true );
	} );

	test( 'Renders data correctly', () => {
		const timeline = mount( <Timeline items={ mockData } /> );

		// Ensure correct divs are loaded.
		expect( timeline.find( '.timeline_no_events' ).length ).toBe( 0 );
		expect( timeline.find( '.woocommerce-timeline' ).length ).toBe( 1 );

		// Ensure groups have the correct number of items.
		expect( timeline.find( '.woocommerce-timeline-group' ).length ).toBe(
			3
		);
		expect(
			timeline
				.find( '.woocommerce-timeline-group ul' )
				.at( 0 )
				.children().length
		).toBe( 1 );
		expect(
			timeline
				.find( '.woocommerce-timeline-group ul' )
				.at( 1 )
				.children().length
		).toBe( 2 );
		expect(
			timeline
				.find( '.woocommerce-timeline-group ul' )
				.at( 2 )
				.children().length
		).toBe( 1 );

		// Ensure dates are correctly rendered.
		expect(
			timeline
				.find( '.woocommerce-timeline-group__title' )
				.first()
				.text()
		).toBe( 'January 22, 2020' );
		expect(
			timeline
				.find( '.woocommerce-timeline-group__title' )
				.last()
				.text()
		).toBe( 'January 17, 2020' );

		// Ensure the correct number of items is rendered.
		expect( timeline.find( '.woocommerce-timeline-item' ).length ).toBe(
			4
		);

		// Ensure hidden timestamps are actually hidden and vice versa.
		expect(
			timeline
				.find( '.woocommerce-timeline-item__timestamp' )
				.at( 2 )
				.text()
		).toBe( '' );
		expect(
			timeline
				.find( '.woocommerce-timeline-item__timestamp' )
				.at( 1 )
				.text()
		).not.toBe( '' );
	} );

	test( 'Empty snapshot', () => {
		const tree = renderer.create( <Timeline /> ).toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	test( 'With data snapshot', () => {
		const tree = renderer
			.create( <Timeline items={ mockData } /> )
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	describe( 'Timeline utilities', () => {
		test( 'Sorts correctly', () => {
			const jan21 = new Date( 2020, 0, 21 );
			const jan22 = new Date( 2020, 0, 22 );
			const jan23 = new Date( 2020, 0, 23 );

			const data = [
				{ id: 0, date: jan22 },
				{ id: 1, date: jan21 },
				{ id: 2, date: jan23 },
			];
			const expectedAsc = [
				{ id: 1, date: jan21 },
				{ id: 0, date: jan22 },
				{ id: 2, date: jan23 },
			];
			const expectedDesc = [
				{ id: 2, date: jan23 },
				{ id: 0, date: jan22 },
				{ id: 1, date: jan21 },
			];

			expect( data.sort( sortByDateUsing( 'asc' ) ) ).toStrictEqual(
				expectedAsc
			);
			expect( data.sort( sortByDateUsing( 'desc' ) ) ).toStrictEqual(
				expectedDesc
			);
		} );

		test( "Empty item list doesn't break sort", () => {
			expect( [].sort( sortByDateUsing( 'asc' ) ) ).toStrictEqual( [] );
		} );

		test( "Single item doesn't change on sort", () => {
			const items = [ { date: new Date( 2020, 0, 1 ) } ];
			expect( items.sort( sortByDateUsing( 'asc' ) ) ).toBe( items );
		} );

		test( 'Groups correctly', () => {
			const jan22 = new Date( 2020, 0, 22 );
			const jan23 = new Date( 2020, 0, 23 );
			const items = [
				{ id: 0, date: jan22 },
				{ id: 1, date: jan23 },
				{ id: 2, date: jan22 },
			];
			const expected = [
				{
					date: jan22,
					items: [
						{ id: 0, date: jan22 },
						{ id: 2, date: jan22 },
					],
				},
				{
					date: jan23,
					items: [ { id: 1, date: jan23 } ],
				},
			];

			expect(
				items.reduce( groupItemsUsing( 'days' ), [] )
			).toStrictEqual( expected );
		} );

		test( "Empty item list doesn't break grouping", () => {
			expect( [].reduce( groupItemsUsing( 'days' ), [] ) ).toStrictEqual(
				[]
			);
		} );

		test( 'Single item grouped correctly', () => {
			const jan22 = new Date( 2020, 0, 22 );
			const items = [ { id: 0, date: jan22 } ];
			const expected = [
				{
					date: jan22,
					items: [ { id: 0, date: jan22 } ],
				},
			];
			expect(
				items.reduce( groupItemsUsing( 'days' ), [] )
			).toStrictEqual( expected );
		} );
	} );
} );
