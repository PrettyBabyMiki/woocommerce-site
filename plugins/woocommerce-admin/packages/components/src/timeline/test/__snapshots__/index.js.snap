// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline Empty snapshot 1`] = `
<div
  className="woocommerce-timeline"
>
  <p
    className="timeline_no_events"
  >
    No data to display
  </p>
</div>
`;

exports[`Timeline With data snapshot 1`] = `
<div
  className="woocommerce-timeline"
>
  <ul>
    <li
      className="woocommerce-timeline-group"
    >
      <p
        className="woocommerce-timeline-group__title"
      >
        January 22, 2020
      </p>
      <ul>
        <li
          className="woocommerce-timeline-item"
        >
          <div
            className="woocommerce-timeline-item__top-border"
          />
          <div
            className="woocommerce-timeline-item__title"
          >
            <div
              className="woocommerce-timeline-item__headline"
            >
              <svg
                className="gridicon gridicons-cross is-error"
                height={16}
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M18.36 19.78L12 13.41l-6.36 6.37-1.42-1.42L10.59 12 4.22 5.64l1.42-1.42L12 10.59l6.36-6.36 1.41 1.41L13.41 12l6.36 6.36z"
                  />
                </g>
              </svg>
              <span>
                string in headline
              </span>
            </div>
            <span
              className="woocommerce-timeline-item__timestamp"
            >
              3:13pm
            </span>
          </div>
          <div
            className="woocommerce-timeline-item__body"
          >
            <span>
              <span>
                span in body
              </span>
            </span>
          </div>
        </li>
      </ul>
      <hr />
    </li>
    <li
      className="woocommerce-timeline-group"
    >
      <p
        className="woocommerce-timeline-group__title"
      >
        January 20, 2020
      </p>
      <ul>
        <li
          className="woocommerce-timeline-item"
        >
          <div
            className="woocommerce-timeline-item__top-border"
          />
          <div
            className="woocommerce-timeline-item__title"
          >
            <div
              className="woocommerce-timeline-item__headline"
            >
              <svg
                className="gridicon gridicons-refresh is-warning"
                height={16}
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M17.91 14c-.478 2.833-2.943 5-5.91 5-3.308 0-6-2.692-6-6s2.692-6 6-6h2.172l-2.086 2.086L13.5 10.5 18 6l-4.5-4.5-1.414 1.414L14.172 5H12c-4.418 0-8 3.582-8 8s3.582 8 8 8c4.08 0 7.438-3.055 7.93-7h-2.02z"
                  />
                </g>
              </svg>
              <span>
                <span>
                  span in headline
                </span>
              </span>
            </div>
            <span
              className="woocommerce-timeline-item__timestamp"
            >
              11:45pm
            </span>
          </div>
          <div
            className="woocommerce-timeline-item__body"
          />
        </li>
        <li
          className="woocommerce-timeline-item"
        >
          <div
            className="woocommerce-timeline-item__top-border"
          />
          <div
            className="woocommerce-timeline-item__title"
          >
            <div
              className="woocommerce-timeline-item__headline"
            >
              <svg
                className="gridicon gridicons-checkmark is-success"
                height={16}
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M9 19.414l-6.707-6.707 1.414-1.414L9 16.586 20.293 5.293l1.414 1.414"
                  />
                </g>
              </svg>
              <span>
                <p>
                  p tag in headline
                </p>
              </span>
            </div>
            <span
              className="woocommerce-timeline-item__timestamp"
            />
          </div>
          <div
            className="woocommerce-timeline-item__body"
          >
            <span>
              <p>
                p element in body
              </p>
            </span>
            <span>
              string in body
            </span>
          </div>
        </li>
      </ul>
      <hr />
    </li>
    <li
      className="woocommerce-timeline-group"
    >
      <p
        className="woocommerce-timeline-group__title"
      >
        January 17, 2020
      </p>
      <ul>
        <li
          className="woocommerce-timeline-item"
        >
          <div
            className="woocommerce-timeline-item__top-border"
          />
          <div
            className="woocommerce-timeline-item__title"
          >
            <div
              className="woocommerce-timeline-item__headline"
            >
              <svg
                className="gridicon gridicons-cross"
                height={16}
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M18.36 19.78L12 13.41l-6.36 6.37-1.42-1.42L10.59 12 4.22 5.64l1.42-1.42L12 10.59l6.36-6.36 1.41 1.41L13.41 12l6.36 6.36z"
                  />
                </g>
              </svg>
              <span>
                undefined body and string headline
              </span>
            </div>
            <span
              className="woocommerce-timeline-item__timestamp"
            >
              1:45am
            </span>
          </div>
          <div
            className="woocommerce-timeline-item__body"
          />
        </li>
      </ul>
      <hr />
    </li>
  </ul>
</div>
`;
